src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -1133,6 +1133,21 @@ namespace InternalHeatGains {
 						Lights( Loop ).FractionReturnAirIsCalculated = ( AlphaName( 6 ) == "YES" );
 					}
 
+					// Set return air node number
+					Lights( Loop ).ReturnNodePtr = 0;
+					std::string retNodeName = "";
+					if ( !lAlphaFieldBlanks( 7 ) ) {
+						if ( LightsObjects( Item ).ZoneListActive ) {
+							ShowSevereError( RoutineName + CurrentModuleObject + "=\"" + Lights( Loop ).Name + "\": " + cAlphaFieldNames( 7 ) + " must be blank when using a ZoneList." );
+							ErrorsFound = true;
+						} else {
+							retNodeName = AlphaName( 7 );
+						}
+					}
+					if ( Lights( Loop ).ZonePtr > 0 ) {
+						Lights( Loop ).ReturnNodePtr = DataZoneEquipment::GetReturnAirNodeForZone( Zone( Lights( Loop ).ZonePtr ).Name, retNodeName );
```

Comment:
The lights won't know.  There still has to be a zone return node and then later (I forget where - in corrector?) there's a decision made whether to put gains on the return node, or if there is zero flow or it's a cycling fan, then put it in the zone heat balance.  This is still currently on a zone-level basis - I need to redo that part to sort them out node-by-node and decide which gains go where.